apps/snake/snake.star

```diff
@@ -1,261 +1,226 @@
-                    snakeDir = ["d", "l", "u"][random.number(0, 2)]
-            if snakeDir == "d":
-                if collideTail(snake, [snakePos[0], snakePos[1] + 1]):
-                    snakeDir = ["l", "u", "r"][random.number(0, 2)]
-            if snakeDir == "l":
-                if collideTail(snake, [snakePos[0] - 1, snakePos[1]]):
-                    snakeDir = ["u", "r", "d"][random.number(0, 2)]
-
-        # get egg
-        if snakePos == egg:
-            egg = newEgg()
-            for _ in range(GROWTH_RATE):
-                tail = [0, 0]
-                tail[0] = snake[-1][0]
-                tail[1] = snake[-1][1]
-                snake.insert(0, tail)
-
-        # render frame
-        frames.append(render_frame(snake, egg))
-
-        # move snake towards egg
-        tail = snake.pop(0)
-        tail[0] = snakePos[0]
-        tail[1] = snakePos[1]
-        if snakeDir == "u":
-            tail[1] = (tail[1] - 1) % HEIGHT
-        elif snakeDir == "r":
-            tail[0] = (tail[0] + 1) % WIDTH
-        elif snakeDir == "d":
-            tail[1] = (tail[1] + 1) % HEIGHT
-        elif snakeDir == "l":
-            tail[0] = (tail[0] - 1) % WIDTH
-        snake.append(tail)
-
-    return frames
-
-def animate(STARTING_SIZE, GROWTH_RATE):
-    frames = playSnake(STARTING_SIZE, GROWTH_RATE)
+    def reset():
+        starting_snake_position_x = int((BOARD_SIZE_X - 1) / 2)
+        snake_position = []
+        for i in range(STARTING_SNAKE_SIZE):
+            snake_position.append(
+                [starting_snake_position_x - i, starting_snake_position_x],
+            )
+
+        snake_direction = SNAKE_DIRECTION_E
+
+        apple_position = new_apple_position(None, snake_position)
+        game_over = False
+        return snake_position, snake_direction, apple_position, game_over
+
+    snake_position, snake_direction, apple_position, game_over = reset()
+
+    for _ in range(MAX_FRAMES):
+        frames.append(
+            render.Column(
+                expanded = True,
+                children = render_board(
+                    game_over,
+                    snake_position,
+                    apple_position,
+                ),
+            ),
+        )
 
-    return render.Animation(children = frames)
+        if not game_over:
+            game_over, snake_position, snake_direction, apple_position = next_move(
+                snake_position,
+                snake_direction,
+                apple_position,
+            )
+        else:
+            snake_position, snake_direction, apple_position, game_over = reset()
 
-def main(config):
-    STARTING_SIZE = 4
-    GROWTH_RATE = 1
+    return render.Animation(children = frames)
 
-    if config.get("STARTING_SIZE"):
-        STARTING_SIZE = int(config.get("STARTING_SIZE"))
-    if config.get("GROWTH_RATE"):
-        GROWTH_RATE = int(config.get("GROWTH_RATE"))
+def main():
+    random.seed(time.now().unix)
```

Comment:
```suggestion
    random.seed(time.now().unix // 15)
```
A little bit more deterministic if you don't mind? Would improve our ability to cache server side.